Content.Shared/Silicon/Systems/SharedSiliconSystem.cs

```diff
@@ -22,65 +22,63 @@ public override void Initialize()
         SubscribeLocalEvent<SiliconComponent, RefreshMovementSpeedModifiersEvent>(OnRefreshMovespeed);
         SubscribeLocalEvent<SiliconComponent, ItemSlotInsertAttemptEvent>(OnItemSlotInsertAttempt);
         SubscribeLocalEvent<SiliconComponent, ItemSlotEjectAttemptEvent>(OnItemSlotEjectAttempt);
+        SubscribeLocalEvent<SiliconComponent, TryingToSleepEvent>(OnTryingToSleep);
     }
 
-    private void OnItemSlotInsertAttempt(EntityUid uid, SiliconComponent component, ref ItemSlotInsertAttemptEvent args)
+    /// <summary>
+    ///     Silicon entities can now also be Living player entities. We may want to prevent them from sleeping if they can't sleep.
+    /// </summary>
+    private void OnTryingToSleep(EntityUid uid, SiliconComponent component, ref TryingToSleepEvent args)
     {
-        if (args.Cancelled)
-            return;
-
-        if (!TryComp<PowerCellSlotComponent>(uid, out var cellSlotComp))
-            return;
+        args.Cancelled = !component.DoSiliconsDreamOfElectricSheep;
```

Comment:
you should not do it like this. Either check if it's canceled first and return, or use `&=` instead of `=`